### ISSUE TITLE
CustomerIOPushNotificationCallback isn’t called if the app is in the background and the notification doesn’t have a deep link

### ISSUE THREAD

**Issue by tfcporciuncula** [customerio/customerio-android#243]
**SDK version:** 3.6.4

**Environment:** Development and Production

**Are logs available?** 

There’s something interesting in the logs: while the app is in the background (open or closed), when I send a notification with deep link, I see this:

>Received message with empty deliveryId
>Handling push message. Bundle: …

But when the notification doesn’t have a deep link, I only see the first message, even though the notification shows up as usual:

>Received message with empty deliveryId

So it’s like [this](https://github.com/customerio/customerio-android/blob/b86e81d9b95a1d3fa1840ba1631d8a9650f787de/messagingpush/src/main/java/io/customer/messagingpush/CustomerIOPushNotificationHandler.kt#L84) is never being called for some reason?

**Describe the bug**

`CustomerIOPushNotificationCallback` isn’t being called if we send a notification without a deep link while the app is in the background (open or closed). It works fine if the app is in the foreground, or if the notification has a deep link.

**To Reproduce**

- Make sure the app is not in the foreground (either kill it or keep it open but not in the foreground)
- Send a notification message without a deep link
- Observe `CustomerIOPushNotificationCallback` won’t be called

**Expected behavior**

`CustomerIOPushNotificationCallback` should always be called.

**Additional context**

I’ve reproduced this on a Pixel 4a running Android 13 so far.

**Comment by Shahroz16** [comment]
Hey @tfcporciuncula, thank you for reaching out and raising this. 

I think I understand what might be happening here, I think you could verify it on your end as well. 

So, when you add the deep link, and try previewing the payload from the dashboard, you would notice its a custom data payload object wrapped in `data`. That means our SDK gets to handle it, 

But when you just have an image in there without the deep link, and if you preview from the dashboard, you would notice the data is under `notification` payload. That means FCM [handles](https://firebase.google.com/docs/cloud-messaging/android/receive) is different for foreground vs background. 

I have raised this with the team, and we are looking into how we can update the flow if you send a push via rich push editor it should always be wrapped under the `data` payload.

**Comment by tfcporciuncula** [comment]
Yup, I’ve verified what you described on my end, thanks! And I agree the SDK should always work with data messages to keep the behavior consistent — I don’t see any good reason to have notification messages (where Firebase is responsible for the rendering) in the mix.

**Comment by jcpsimmons** [comment]
Closing this out as stale - it dates back to 2023 and the push handling internals have been substantially reworked since. If you're still hitting this on the latest SDK, please reopen or file a fresh issue with details and we'll dig in.
